core/packages/contracts/contracts/utils/Bitfield.sol

```diff
@@ -48,87 +41,103 @@ library Bitfield {
         bitfield = new uint256[](prior.length);
         uint256 found = 0;
 
-        for (uint256 i = 0; found < n; i++) {
+        for (uint256 i = 0; found < n;) {
             bytes32 randomness = keccak256(abi.encodePacked(seed, i));
```

Comment:
You can still optimize this hashing operation by using this assembly here: https://github.com/Snowfork/snowbridge/blob/a17e18c6425c4389ef2bcbc27113663c3eac95b3/core/packages/contracts/contracts/BeefyClient.sol#L371

Also this guide may help add context to what that assembly is doing: https://docs.soliditylang.org/en/v0.8.15/internals/layout_in_memory.html